### ISSUE TITLE
Active Record models with different database connections

### ISSUE THREAD

**Issue by senny** [DataDog/dd-trace-rb#238]
Our application uses Active Record to connect to more than one database. The connections are defined on a per model bases. Now in DataDog APM all queries are shown under the same database service. Is it possible to split them by this connection somehow? It would be nice if we had the throughput (queries per second) and latency metrics split per database.

**Comment by senny** [comment]
I was hoping that #270 would allow us to split queries by using a span processor. Unfortunately, the `rails.database.name` tag is always set to the default connection, regardless of the query that's being annotated.

It's a bit confusing as the name suggests that it is possible to tell different databases on the same Rails app apart.

**Comment by delner** [comment]
Ahhhh yeah, this is a good point. I think having the ability to configure individual ActiveRecord connections might be the best way to go here. Maybe something like:

```ruby
Datadog.configure do
  use :rails
end

Datadog.configure(Foo.connection, service_name: 'foo-db')
Datadog.configure(Bar.connection, service_name: 'bar-db')
```

...or something similar. ActiveRecord needs a bit more love; I'd be happy to take a deeper dive into this when we tackle it.

**Comment by senny** [comment]
@delner yes, something like this would be perfect. Since it's in an initializer it should ideally not open the connection. Maybe it's possible just by using the name of the connection specification in `database.yml`.

**Comment by delner** [comment]
Hey @senny , we're going to look at this in a couple of weeks. I'll keep you updated.

**Comment by lattwood** [comment]
@delner anything to update?

**Comment by delner** [comment]
@lattwood Not yet, other than it's still queued up on our roadmap. Will let you know what develops.

**Comment by delner** [comment]
FWIW @senny abd @lattwood #383 should tag `sql.active_record` events with the correct connection information now. Still don't have support for tracking those different DB connections as different services yet, which is something we'll have to add.

**Comment by delner** [comment]
Hey @senny and @lattwood , #404 is currently in development and should resolve this. Feel free to check it out!

**Comment by delner** [comment]
We made another PR #451 that improves upon the configuration behavior; this is currently in review pending some other changes to our configuration API.
